### ISSUE TITLE
Feature Request: Automatically account for holidays

### ISSUE THREAD

**Issue by ciejepro** [bruxy70/Garbage-Collection#54]
Please consider adding an automatic accounting for national holidays in the country of use.
For example, typically in the US, trash/recycling would be picked up one day later when a US holiday such as Christmas falls on a garbage collection day.

This is true in 2019, Christmas falls on Wednesday, so our local Garbage collection (normally Wednesday) will now happen on Thursday.

Adding a national holiday component based on country and a simple checkbox for the integration to observe holidays would be awesome.  This avoid having to determine and add the holidays by individual date for an entire year, and also adding the dates to include, since the tool does not automatically account for it to happen one day later.

**Comment by bruxy70** [comment]
If it was the next day everywhere, that would be easy. I have looked into that some time ago. Turns out, there are so many different variants for when the next collection would happen. As well as not all holidays are automatically excluded, for example in my village - 25th will be a collection as normal, but 1/1 they will move it to Thursday. So after some thought, I went for a manual inclusion/exclusion.

**Comment by ciejepro** [comment]
Agreed, but there are still the include dates.  I am shocked that there do 
u of be anywhere in thebus with christmas day collection but anything is 
possible.

If this could be an optional feature that would be awesome. Nit has been 
great so far this would just add the its value


CJ Ramseyer
Production/Technical Director
CieJe Productions, LLC
ciejepro.com
cj@ciejepro.com
(203) 690-4969

**Comment by bruxy70** [comment]
So, what I plan to do is, to use the `holidays` package. For the US, it renders the following list of US holidays for next year:
```
2020-01-01 New Year's Day
2020-01-20 Martin Luther King, Jr. Day
2020-02-17 Washington's Birthday
2020-05-25 Memorial Day
2020-07-04 Independence Day
2020-07-03 Independence Day (Observed)
2020-09-07 Labor Day
2020-10-12 Columbus Day
2020-11-11 Veterans Day
2020-11-26 Thanksgiving
2020-12-25 Christmas Day
```
The collections days would be a day later (regardless of if this comes on the weekend). But if this comes to another bank holiday it would be moved further to another day. Is that correct?

I am asking you to validate the logic before I do that, since I have no idea - they do not do that here. Suppose because they have different jobs scheduled for different days. So if they do the place I live on Wednesday, they do some other place (or different waste type) on Thursday. So if they did what they do in the US, they would have to make up for 2 and sometimes 3 days in 1 day. I wonder how they do that. Anyhow, let me know.

**Comment by ciejepro** [comment]
Exactly.  Then just making the observation of this method an option for 
each collection type.  I think thus is awesome.

Cheers,
CJ

CJ Ramseyer
Production/Technical Director
CieJe Productions, LLC
ciejepro.com
cj@ciejepro.com
(203) 690-4969

**Comment by bruxy70** [comment]
Included in release 2.11. Check it out.

**Comment by ciejepro** [comment]
I just tried this and I like the way you implemented it.  A choice to turn 
it on by country.

Nice job!  Thanks for the quick implementation.


CJ Ramseyer
Production/Technical Director
CieJe Productions, LLC
ciejepro.com
cj@ciejepro.com
(203) 690-4969
